vm/src/types/slot.rs

```diff
@@ -30,7 +30,10 @@ macro_rules! atomic_func {
 #[derive(Default)]
 #[non_exhaustive]
 pub struct PyTypeSlots {
-    pub name: PyRwLock<Option<String>>, // tp_name, not class name
+    /// # Safety
+    /// For static types, always safe.
+    /// For heap types, `__name__` must alive
+    pub(crate) name: AtomicCell<&'static str>, // tp_name with <module>.<class> for print, not class name
```

Comment:
`slots.name` is now atomic cell. And the value itself is unsafe for heaptypes. instead of direct accessing, `slot_name()` will be safe.
So this field is not pub anymore.

In fact, this field even does not need to be `AtomicCell` anymore. UnsafeCell?